File: YuGiOh2/Cards/C52097679.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 右手持盾左手持剑
    /// </summary>
    public class C52097679
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown) ||
                player.Enemy.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown);
        }

        public static void ProcessEffect(Player player)
        {
            Change(player);
            Change(player.Enemy);
        }

        private static void Change(Player player)
        {
            foreach (var monster in player.Field.MonsterFields)
            {
                if (monster == null || monster.Status.FaceDown)
                    continue;

                int tmp = monster.ATK;
                monster.ATK = monster.DEF;
                monster.DEF = tmp;
            }
        }
    }
}

File: YuGiOh2/Scripts/C52097679.lua
Type = 0;
--右手持盾左手持剑
function checkIfAvailable(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    return false;
end

function processEffect(player)
    change(player);
    change(player.Enemy);
end

function change(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            local tmp = monster.ATK;
            monster.ATK = monster.DEF;
            monster.DEF = tmp;
        end
    end
end